.github/workflows/adaptertest.yaml

```diff
@@ -0,0 +1,168 @@
+name: Adapter Test
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "adapters/*.ts"
+
+  issue_comment:
+    types: [created]
+
+jobs:
+  adapter-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Deno
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: v2.x
+
+      - name: Parse test address
+        id: parse
+        run: |
+          # Default address for PR events
+          DEFAULT_ADDRESS="0x69155e7ca2e688ccdc247f6c4ddf374b3ae77bd6"
+
+          # Check if this is a comment event
+          if [ "${{ github.event_name }}" = "issue_comment" ]; then
+            COMMENT_BODY="${{ github.event.comment.body }}"
+            # Extract address from /test <address> command
+            ADDRESS=$(echo "$COMMENT_BODY" | grep -oE '/test\s+0x[a-fA-F0-9]{40}' | grep -oE '0x[a-fA-F0-9]{40}')
+            if [ -z "$ADDRESS" ]; then
+              echo "Invalid or missing address in comment"
+              exit 0
+            fi
+          else
+            ADDRESS="$DEFAULT_ADDRESS"
+          fi
+
+          # Validate address format
+          if [[ ! "$ADDRESS" =~ ^0x[a-fA-F0-9]{40}$ ]]; then
```

Comment:
i am not sure if you want to validate here because it is already done in test.ts, will make it extra work once we support more address formats as well